src/scheduler/node_info.c

```diff
@@ -1678,15 +1678,9 @@ update_node_on_end(node_info *ninfo, resource_resv *resresv, char *job_state)
 		remove_resresv_from_array(ninfo->run_resvs_arr, resresv);
 	}
 
-	/* Since we are removing work from a node, it'll free up some cpus.  This
-	 * means it'll go from exclusive or shared to free.  If the node is busy
-	 * removing work will not make it leave the busy state until the loadave
-	 * has lowered.  This will take time.
-	 */
-	if (!ninfo->is_busy) {
-		set_node_info_state(ninfo, ND_free);
-	}
-	else if (is_excl(resresv->place_spec, ninfo->sharing)) {
+	if (ninfo->is_job_busy)
+		remove_node_state(ninfo, ND_jobbusy);
```

Comment:
Can you please explain the if condition change? Why from `!ninfo->is_busy` to `info->is_job_busy`? Thanks.

Reply:
They are two different states.  The job_busy state means all the cpus are busy.  The busy state means the load average of the node has gone above its max_load.  This is really one of the main parts of the whole PR.  Before if the node state was not busy, it'd set the state to free.  If the node state was stale or down, it would be set to free.  The new code checks if the node is in state job_busy and if so, remove the state bit.  This will leave other states like stale set.